File: src/main/java/io/github/slimefunguguproject/bump/utils/ValidateUtils.java
package io.github.slimefunguguproject.bump.utils;

import javax.annotation.Nullable;

import com.google.common.base.Preconditions;

import org.bukkit.inventory.ItemStack;

import lombok.experimental.UtilityClass;

/**
 * This utility class holds validation methods.
 *
 * @author ybw0014
 */
@UtilityClass
public final class ValidateUtils {
    /**
     * Validate the given {@link ItemStack} is not {@code null} or an air item.
     *
     * @param itemStack The {@link ItemStack} to be validated.
     *
     * @return if the {@link ItemStack} is valid.
     */
    public static boolean noAirItem(@Nullable ItemStack itemStack) {
        try {
            Preconditions.checkArgument(itemStack != null, "ItemStack cannot be null");
            Preconditions.checkArgument(!itemStack.getType().isAir(), "ItemStack cannot be empty");
            return true;
        } catch (IllegalArgumentException ex) {
            return false;
        }
    }
}

File: src/main/kotlin/io/github/slimefunguguproject/bump/utils/items/ValidateUtils.kt
package io.github.slimefunguguproject.bump.utils.items

import org.bukkit.inventory.ItemStack

object ValidateUtils {
    /**
     * Validate the given [ItemStack] is not `null` or an air item.
     *
     * @param itemStack The [ItemStack] to be validated.
     *
     * @return if the [ItemStack] is valid.
     */
    fun noAirItem(itemStack: ItemStack?) = itemStack != null && !itemStack.type.isAir

    /**
     * Validate the given [ItemStack] is not `null`, not an air item, and has item meta.
     *
     * @param itemStack The [ItemStack] to be validated.
     *
     * @return if the [ItemStack] is valid.
     */
    fun noAirAndHasItemMeta(itemStack: ItemStack?) = noAirItem(itemStack) && itemStack!!.hasItemMeta()
}
